sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+## API Report File for "@azure/eventgrid"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AzureKeyCredential } from '@azure/core-auth';
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { RestResponse } from '@azure/core-http';
+
+export { AzureKeyCredential }
+
+// @public
+export interface CloudEvent<T> {
+    [keyof: string]: any;
+    data?: T;
+    datacontenttype?: string;
+    dataschema?: string;
+    id: string;
+    source: string;
+    subject?: string;
+    time?: Date;
+    type: string;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventData {
+    action?: string;
+    id?: string;
+    target?: ContainerRegistryArtifactEventTarget;
+    timestamp?: Date;
+}
+
+// @public
+export interface ContainerRegistryArtifactEventTarget {
+    digest?: string;
+    mediaType?: string;
+    name?: string;
+    repository?: string;
+    size?: number;
+    tag?: string;
+    version?: string;
+}
+
+// @public
+export type ContainerRegistryChartDeletedEventData = ContainerRegistryArtifactEventData & {};
```

Comment:
The empty object type isn't necessary from a type perspective. Is this done to make tooling report `ContainerRegistryChartDeletedEventData` instead of `ContainerRegistryChartDeletedEventData`?

Reply:
> The empty object type isn't necessary from a type perspective. Is this done to make tooling report `ContainerRegistryChartDeletedEventData` instead of `ContainerRegistryChartDeletedEventData`?

(I think you mean "instead of `ContainerRegistryArtifactEventData`", so I'm going to respond as if that is true).

Ultimately, these types were auto-generated from the swagger and the way the swagger is authored puts us into this place, [I think because of the use of `allOf` so they don't need to duplicate a bunch of data](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/eventgrid/data-plane/Microsoft.ContainerRegistry/stable/2018-01-01/ContainerRegistry.json#L10-L17).

I can see if there's transforms we can do on the Auto Rest side to clean things up, since as you say the extra types don't add value here.

Reply:
We decided to not try to fight the swagger here, so we'll keep this as is for now.